### PR TITLE
[Enhancement] Complete the error message when selecting an Iceberg table created by spark2

### PR DESCRIPTION
**Pull request by mxdzs0612**
## What type of PR is this：
- [ ] bug
- [ ] feature
- [x] enhancement
- [ ] refactor
- [ ] others

## Which issues of this PR fixes ：

Fixes #

## Problem Summary(Required) ：

Iceberg tables created with spark2 by default configurations don't have the property 'metadata_location', which is necessary when selected using Hive catalog. Trino can neither select such table, so I suppose it reasonable and just throw an exception with reasons in detail.

MySQL client will crash when executing `show tables` after using a database with such tables. I'll try to solve it later in a separated PR.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] I have added user document for my new feature or new function

### CODE REVIEW

**Comment by mxdzs0612** [comment]
@caneGuy @stephen-shelby PTAL

**Comment by mxdzs0612** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 0 / 2 (00.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/external/iceberg/hive/HiveTableOperations.java|0|2|00.00%|[81, 82]|

File: fe/fe-core/src/main/java/com/starrocks/external/iceberg/hive/HiveTableOperations.java [hunk 1]
```diff
@@ -77,6 +78,10 @@ protected void doRefresh() {
             validateTableIsIceberg(table, fullName);
 
             metadataLocation = table.getParameters().get(METADATA_LOCATION_PROP);
+            if (null == metadataLocation || metadataLocation.length() == 0) {
```
Comment (imay) [inline fe/fe-core/src/main/java/com/starrocks/external/iceberg/hive/HiveTableOperations.java:81]: you can use Strings.isNullOrEmpty() in guava

**Comment by mxdzs0612** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 0 / 2 (00.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/external/iceberg/hive/HiveTableOperations.java|0|2|00.00%|[82, 83]|

**Review by stephen-shelby — APPROVED** [review]
LGTM

**Review by imay — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by mxdzs0612** [comment]
@Mergifyio backport branch-2.4

**Pull request merged**